internal/ingress/annotations/annotations.go

```diff
@@ -76,12 +78,14 @@ type Ingress struct {
 	ExternalAuth         authreq.Config
 	HealthCheck          healthcheck.Config
 	Proxy                proxy.Config
+	ProxyPass            proxypass.Config
 	RateLimit            ratelimit.Config
 	Redirect             redirect.Config
 	Rewrite              rewrite.Config
 	SecureUpstream       secureupstream.Config
 	ServerSnippet        string
 	ServiceUpstream      bool
+	ServiceNameAsVhost   bool
```

Comment:
Instead of introducing a new annotation type for `service-name-as-vhost`, could it somehow be consolidated with `upstream-vhost`?

Reply:
It could - I'm not sure whether you mean adding the `service-name-as-host` annotation onto the existing `upstreamVhost` type using a `Config` struct (the same way other multi-annotation types are handled), or simply removing the `service-name-as-vhost` annotation and replacing `{{ if $location.ServiceUpstreamAsVhost }}` with something like `{{ if (and (not (empty $location.UpstreamVhost)) (not (empty $location.ProxyPass.Address))) }}` in the nginx template to replicate the functionality, but either one is doable.

Reply:
@antoineco Turns out I actually don't need the `service-name-as-vhost` annotation, as it's easier to integrate with linkerd by adding a custom header (via custom nginx template or configuration snippet) and then having linkerd route on that instead of trying to change the `Host` header.